pkg/gardenlet/controller/shoot/care/reconciler.go

```diff
@@ -107,9 +107,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		gardencorev1beta1.ShootAPIServerAvailable,
 		gardencorev1beta1.ShootControlPlaneHealthy,
 		gardencorev1beta1.ShootObservabilityComponentsHealthy,
-		gardencorev1beta1.ShootEveryNodeReady,
-		gardencorev1beta1.ShootSystemComponentsHealthy,
 	}
+
+	if !v1beta1helper.IsWorkerless(shoot) {
+		conditionTypes = append(conditionTypes,
+			gardencorev1beta1.ShootEveryNodeReady,
+			gardencorev1beta1.ShootSystemComponentsHealthy,
```

Comment:
There might still be system components (not deployments etc., but other resources (RBAC, etc.)). Perhaps we should keep it?